### PR TITLE
Weighting wall attributes

### PR DESCRIPTION
**Pull request by leifurhauks**
The way it works is that the "weight" for each value is the number of times that value is appears in the random drawing box, like buying multiple tickets in a raffle.

### CODE REVIEW

**Comment by leifurhauks** [comment]
Also, for context-dependent weights, we could have multiple weight tables and pass the relevant table for the current context to the random_draw function.

File: sk8border.p8 [hunk 1]
```diff
@@ -578,11 +596,25 @@ function _init()
...
   palt(0,false)
-  palt(7,true)
-  
+  palt(7,true) 
+
+  wall_height_drawing_box = {}
+  wall_width_drawing_box = {}
+
+  fill_drawing_box(wall_height_weights, wall_height_drawing_box)
+  fill_drawing_box(wall_width_weights, wall_width_drawing_box)
```
Comment (benwiley4000) [inline sk8border.p8]: Does this mean that when the game starts we have one of each wall height instead of random?

Comment (benwiley4000) [reply sk8border.p8]: I think I might just be misunderstanding the code

**Comment by leifurhauks** [comment]
The drawing box is like a bucket full of raffle tickets (except in this
analogy, we're putting the winning ticket back into the bucket between
drawings). So if there's one "ticket" for each wall height/width, there's
an equal chance of drawing each wall height/width. When we decide how we
want to distribute the weights, we'll increase the number of tickets for
values we want to draw more often. I tested it by increasing the weight for
height 5 in wall_height_weights to a much higher number like 3 or 4, and
seeing that there were a lot more short walls when I played the game.

**Comment by benwiley4000** [comment]
Makes sense!! Please merge when you're comfortable with this, I think it looks great.

**Pull request merged**